src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let result = this[0];
+    let i = 1;
+
```

Comment:
Everything is well done. interesting solution. If you go deeper, you can add checks for the incoming value. 
if (typeof initialValue === 'undefined') {
      throw new TypeError}